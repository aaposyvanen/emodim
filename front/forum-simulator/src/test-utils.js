import React from 'react'
import { render as rtlRender } from '@testing-library/react'
import { createStore } from 'redux'
import { Provider } from 'react-redux'

import createRootReducer from './reducer'

// This can be used instead of rtlRender to easily render connected (Redux) components.
// Can be especially useful when children of the tested component are connected.
// There might be a better way, (e.g. shallow rendering) but at least this works.
// Inspiration for this approach:
// https://redux.js.org/recipes/writing-tests#connected-components

function render(
    ui,
    {
        initialState,
        store = createStore(createRootReducer, initialState),
        ...renderOptions
    } = {}
) {
    function Wrapper({ children }) {
        return <Provider store={store}>{children}</Provider>
    }
    return rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
}

// re-export everything
export * from '@testing-library/react'
// override render method
export { render }
