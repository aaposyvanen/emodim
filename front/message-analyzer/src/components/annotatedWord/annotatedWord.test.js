import React from 'react';
import { render } from "../../test-utils";
import AnnotatedWord from "./annotatedWord";

test("AnnotatedWord renders correctly with neutral rating", () => {
    const wordData = {
        type: "WORD",
        word: "test",
        arousal: 0,
        dominance: 0,
        valence: 0
    }
    const { container } = render(<AnnotatedWord wordData={wordData} annotations />);
    expect(container).toMatchSnapshot();
});

test("AnnotatedWord renders correctly with non-neutral rating without annotations", () => {
    const wordData = {
        type: "WORD",
        word: "test",
        arousal: 1,
        dominance: 0,
        valence: 1
    }
    const { container } = render(<AnnotatedWord wordData={wordData} />);
    expect(container).toMatchSnapshot();
});

test("AnnotatedWord renders correctly with negative valence", () => {
    const wordData = {
        type: "WORD",
        word: "test",
        arousal: 0,
        dominance: 0,
        valence: -1
    }
    const { container } = render(<AnnotatedWord wordData={wordData} annotations />);
    expect(container).toMatchSnapshot();
});

test("AnnotatedWord renders correctly with positive valence", () => {
    const wordData = {
        type: "WORD",
        word: "test",
        arousal: 0,
        dominance: 0,
        valence: 1
    }
    const { container } = render(<AnnotatedWord wordData={wordData} annotations />);
    expect(container).toMatchSnapshot();
});

test("AnnotatedWord renders correctly with high arousal", () => {
    const wordData = {
        type: "WORD",
        word: "test",
        arousal: 1,
        dominance: 0,
        valence: 0
    }
    const { container } = render(<AnnotatedWord wordData={wordData} annotations />);
    expect(container).toMatchSnapshot();
});

test("AnnotatedWord renders correctly with punctuation word", () => {
    const wordData = {
        type: "PUNCTUATION",
        word: ","
    }
    const { container } = render(<AnnotatedWord wordData={wordData} />);
    expect(container).toMatchSnapshot();
});

test("AnnotatedWord renders correctly with whitespace word", () => {
    const wordData = {
        type: "WHITESPACE",
        word: " "
    }
    const { container } = render(<AnnotatedWord wordData={wordData} />);
    expect(container).toMatchSnapshot();
});

test("AnnotatedWord renders correctly with line end word", () => {
    const wordData = {
        type: "UNKNOWN",
        word: "\\n\\n"
    }
    const { container } = render(<AnnotatedWord wordData={wordData} />);
    expect(container).toMatchSnapshot();
});

test("AnnotatedWord renders correctly with unknown word", () => {
    const wordData = {
        type: "UNKNOWN",
        word: "#"
    }
    const { container } = render(<AnnotatedWord wordData={wordData} />);
    expect(container).toMatchSnapshot();
});
