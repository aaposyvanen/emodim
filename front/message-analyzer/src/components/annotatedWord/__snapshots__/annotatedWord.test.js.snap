// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnnotatedWord renders correctly with high arousal 1`] = `
<div>
  <span
    class="word arousal2 valence0"
  >
    test
  </span>
</div>
`;

exports[`AnnotatedWord renders correctly with line end word 1`] = `
<div>
  <div
    class="line-break"
  >
    

  </div>
</div>
`;

exports[`AnnotatedWord renders correctly with negative valence 1`] = `
<div>
  <span
    class="word arousal0 valence-2"
    title="T채st채 sanasta tunnistettiin vahvoja tunteita."
  >
    test
  </span>
</div>
`;

exports[`AnnotatedWord renders correctly with neutral rating 1`] = `
<div>
  <span
    class="word arousal0 valence0"
  >
    test
  </span>
</div>
`;

exports[`AnnotatedWord renders correctly with non-neutral rating without annotations 1`] = `
<div>
  <span
    class="word arousal0 valence0"
  >
    test
  </span>
</div>
`;

exports[`AnnotatedWord renders correctly with positive valence 1`] = `
<div>
  <span
    class="word arousal0 valence2"
    title="T채st채 sanasta tunnistettiin vahvoja tunteita."
  >
    test
  </span>
</div>
`;

exports[`AnnotatedWord renders correctly with punctuation word 1`] = `
<div>
  <span
    class="punctuation"
  >
    ,
  </span>
</div>
`;

exports[`AnnotatedWord renders correctly with unknown word 1`] = `<div />`;

exports[`AnnotatedWord renders correctly with whitespace word 1`] = `
<div>
  <span
    class="whitespace"
  >
     
  </span>
</div>
`;
